### ISSUE TITLE
`IEntityType` is replaced by `ITypeBase` but I loose certain methods that I need for EfCore.SchemaCompare library

### ISSUE THREAD

**Issue by JonPSmith** [dotnet/efcore#34594]
I used `IEntityType` to get lots of information when comparing the DbContext's `Model` with the database in my [`EfCore.SchemaCompare`](https://github.com/JonPSmith/EfCore.SchemaCompare) library. I see that `IEntityType` is obsolete and I changed to `ITypeBase`. However, the `ITypeBase` doesn't have certain methods that I need. They are:

- `GetForeignKeys()`
- `FindPrimaryKey()`
- `Model.GetIndexes()`

Also the `IProperty`'s `DeclaringEntityType` is obsolete and I can't find a method to tell me if an entity is `Owned`. The old code used `property.DeclaringEntityType.IsOwned()`.

### Include your code

The [Stage1Comparer](https://github.com/JonPSmith/EfCore.SchemaCompare/blob/master/EfSchemaCompare/Internal/Stage1Comparer.cs) contains code to compare the DbContext's `Model` against the database data, provided by the EF Core's `Scaffolder`.

- **Table/View exists**: That a table or view that an entity class is mapped exists. This checks both table/view name and schema name
- **Property/Column:**  exists, database type (including size and precision), nullability, computed column (including persistence), column default value, when updated (e.g. column is updated `OnAdd` for a int primary key which is provided by the database)
- **Primary key:** SQL constraint name, properties
- **Foreign keys:** SQL constraint name, Delete behavior, properties
- **Indexes:** SQL constraint name, Unique/not unique, properties

### Include provider and version information

EF Core version: 8.0.2
Database provider: all
Target framework: .NET 8.0
Operating system: Windows
IDE: Visual Studio 2022 17.11.1

**Comment by roji** [comment]
Hey @JonPSmith 👋 

When you get an ITypeBase, you can simply check if it happens to be an IEntityType (which extends ITypeBase), and if so, you can access all the APIs you used to on that. The alternative is that you have an IComplexType, which e.g. by definition has no primary key, or foreign keys (at least not at the moment).

> Also the IProperty's DeclaringEntityType is obsolete and I can't find a method to tell me if an entity is Owned. The old code used property.DeclaringEntityType.IsOwned().

Yep - only entity types can be owned; complex types are implicitly part of their containing entity type (or complex type, if they're nested). So like above, you can first check if the ITypeBase is an IEntityType before then checking if it's owned.

**Comment by JonPSmith** [comment]
Thanks @roji,

I have successfully moved from `IEntityType` to `ITypeBase` - used the `ITypeBase`'s `ContainingEntityType` property to get the `IEntityType`.

I have provide of the changes I had to  in case that others developers might be interested. They are:

- I replaced the use of `IEntityType` with the `ITypeBase`. See [EF Core 8.0, breaking-changes](https://learn.microsoft.com/en-us/ef/core/what-is-new/ef-core-8.0/breaking-changes#new-behavior-5) for why I had do this.
- I used the `ITypeBase`'s `ContainingEntityType` property to get the `IEntityType` version, which has extra Metadata that I needed, e.g. `entityType.GetForeignKeys()` is now `typeBase.ContainingEntityType.GetForeignKeys()`.
- The Net 8's `IProperty` has some changes, i.e. 
   - old version = `property.IsPrimaryKey()`
   - new version = `property.FindContainingPrimaryKey()`
